python/sglang/srt/managers/detokenizer_manager.py

```diff
@@ -46,10 +46,19 @@ async def handle_loop(self):
                 # Trim stop str
                 # TODO(lmzheng): handle the case where multiple stop strs are hit
                 for i in range(len(output_strs)):
+                    truncate_len = len(output_tokens[i])
                     if recv_obj.hit_stop_str[i] is not None:
                         pos = output_strs[i].find(recv_obj.hit_stop_str[i])
                         if pos != -1:
                             output_strs[i] = output_strs[i][:pos]
+                            truncate_len = truncate_len-len(self.tokenizer.encode(recv_obj.hit_stop_str[i]))
+                    
+                    # cur_output_tokens = output_tokens[i][:truncate_len]
+                    cur_output_tokens = output_tokens[i]
+                    if cur_output_tokens[-1] == self.tokenizer.eos_token_id:
+                        recv_obj.meta_info[i]['output_tokens'] = cur_output_tokens[:-1]
+                    else:
+                        recv_obj.meta_info[i]['output_tokens'] = cur_output_tokens
```

Comment:
Can you explain why this is needed?

Reply:
Oh sry I just saw it. I think this piece of code just tries to make sure the token output is aligned with text output. If I remember correctly, when I tested last time, if the last token is EOS, the text output does not include the `<|endoftext|>` token while the `<|endoftext|>` token id will be included in `output_tokens` without this code. But perhaps I am hallucinating on this since I did the test a while ago...